Added swiftformat configuration

ckaik opened the pull request:
I've created basic swiftformat + swiftlint configs. Got both of them form [here](https://github.com/gurhub/dotfiles/blob/master) but made some modifications.

Both can be used from the `Makefile`, instructions are in the README.

It didn't really change all that much. Mostly there's more wrapping for initializers, cause some of them are _really_ long. Also it does remove the File Header, that can be disabled though, I just enabled that option because I personally prefer my files without the headers.

One thing that can't be fixed using these tools would be the naming convention of parameters. Snake case is being used a lot right now but camel case would make more sense for Swift. That is a big undertaking though.

fishcharlie commented on Makefile:
```diff
@@ -0,0 +1,5 @@
+lint:
+	swiftlint --fix
```
What if we were to have a `lint` and a `lint-fix` or something like that?

ckaik replied:
Thought about that, shouldn't be an issue.

fishcharlie commented on Makefile:
```diff
@@ -0,0 +1,5 @@
+lint:
+	swiftlint --fix
+
+format:
+	swiftformat .
```
Does this automatically fix invalid formatting? Or does it just throw an error if invalid formatting is detected?

fishcharlie reviewed (changes requested):
Overall this looks good. Couple of code level comments.

One more high level thing. I think it'd be useful to have a GitHub action that runs both the lint and format, but does not actually fix anything.

That way it ensures that any PR made actually conforms to the style before merging.

ckaik commented:
I've added a new `lint-fix` target. The existing `lint` target now just lists the issues.

fishcharlie commented:
@ckaik If we do a `--dryrun` option for `make format` and rename the existing one to `make format-fix`. Then we should be able to add a GitHub Action here. Something like this maybe?

```yml
name: CI

on:
  push:
  pull_request:

jobs:
  lint:
    runs-on: macos-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      - name: Lint code
        run: make lint
      - name: Format code
        run: make format
```

ckaik commented:
Honestly I haven't used Github actions yet so I'm not familiar. It looks fine though ;) I did add the dryrun option to the format target.

Not sure if the swiftformat output is _really_ useful though. It only shows the rules would have been applied and which files they would apply to, no specifics.

Lint is more useful.

fishcharlie commented:
@ckaik I think it depends on the exit code. If the exit code is 0, it'll pass, and if it's anything else it'll fail.

So I guess we need to verify that `swiftformat` actually does return an exit code >0 on dryrun if changes need to be applied.

I can work on the GitHub action piece later.

ckaik commented:
Made another update. The `--dryun` argument doesn't return non-zero if it fails. However the `--lint` option _does_. So I added that instead.

Though now I'm not sure if both swiftformat _and_ swiftlint are needed

fishcharlie commented:
@ckaik The thing I was most concerned about was enforcing consistency with tabs vs spaces. So if we need to remove one to simplify things, that would be fine by me, so long as that main objective is achieved.

ckaik commented:
Then I would propose to remove `swiftlint`, also:

* change the `format` target to run swiftformat like before, fixing issues
* change the lint target to execute `swiftformat --lint .` - that'll return an error code if it doesn't pass

Tab indentation is a rule in swiftformat.

ckaik commented:
Made the changes.

* Removed `.swiftlint.yml`
* `make format` now runs `swiftformat .`, formatting all code
* `make lint` now runs `swiftformat --lint .`, checking the formatting rules for all code
* Adjusted README

So your proposed github action would now only need to run `make lint`.

ckaik commented:
One more update, I just reverted the actual reformatting of the whole project.

I believe this should be done after merging. 
This will make the PR easier to digest.
